### PR TITLE
Adding support for AMA on Arc

### PR DESCRIPTION
**Pull request by nidhanda**
Adding extension support for AMA on Arc. Will be running a separate arc daemon to keep track of the valid key/token files for making the MSI call to Arc's hybrid service. 

**Work remaining -** Adding retry logic for arc watcher and making it more resilient.

### CODE REVIEW

File: AzureMonitorAgent/agent.py [hunk 1]
```diff
@@ -668,16 +680,122 @@ def metrics():
...
+    """
+    This is needed to override mdsd's syslog permissions restriction which prevents mdsd 
+    from reading temporary key files that are needed to make https calls to get an MSI token for arc during onboarding to download amcs config
+    This method spins up a process that will continuously keep refreshing that particular file path with valid keys
+    So that whenever mdsd needs to refresh it's MSI token, it is able to find correct keys there to make the https calls
+    """
+    # check every 25 seconds
+    sleepTime =  25
+
+    # sleep before starting the monitoring.
```
Comment (simathih) [inline AzureMonitorAgent/agent.py:760]: verify this

**Review by simathih — APPROVED** [review]

File: AzureMonitorAgent/agent.py (hunk 1 above)
Comment (nidhanda) [reply AzureMonitorAgent/agent.py:760]: Is this for that whole verification on RHEL or just this specific code block?

Comment (simathih) [reply AzureMonitorAgent/agent.py:760]: both :)

Comment (nidhanda) [reply AzureMonitorAgent/agent.py:760]: Tested both on RHEL. Works as expected

**Pull request merged**